kaldb/src/main/java/com/slack/kaldb/recovery/RecoveryService.java

```diff
@@ -217,41 +231,49 @@ private void handleRecoveryTaskAssignment(RecoveryNodeMetadata recoveryNodeMetad
   @VisibleForTesting
   boolean handleRecoveryTask(RecoveryTaskMetadata recoveryTaskMetadata) {
     LOG.info("Started handling the recovery task: {}", recoveryTaskMetadata);
-    try {
-      RecoveryChunkManager<LogMessage> chunkManager =
-          RecoveryChunkManager.fromConfig(
-              meterRegistry,
-              searchMetadataStore,
-              snapshotMetadataStore,
-              kaldbConfig.getIndexerConfig(),
-              blobFs,
-              kaldbConfig.getS3Config());
-      // Ingest data in parallel
-      LogMessageTransformer messageTransformer =
-          INDEXER_DATA_TRANSFORMER_MAP.get(kaldbConfig.getIndexerConfig().getDataTransformer());
-      LogMessageWriterImpl logMessageWriterImpl =
-          new LogMessageWriterImpl(chunkManager, messageTransformer);
-      KaldbKafkaConsumer kafkaConsumer =
-          KaldbKafkaConsumer.fromConfig(
-              makeKafkaConfig(kaldbConfig.getKafkaConfig(), recoveryTaskMetadata.partitionId),
-              logMessageWriterImpl,
-              meterRegistry);
-      kafkaConsumer.prepConsumerForConsumption(recoveryTaskMetadata.startOffset);
-      kafkaConsumer.consumeMessagesBetweenOffsetsInParallel(
-          KaldbKafkaConsumer.KAFKA_POLL_TIMEOUT_MS,
-          recoveryTaskMetadata.startOffset,
-          recoveryTaskMetadata.endOffset);
-      // Wait for chunks to upload.
-      boolean success = chunkManager.waitForRollOvers();
-      // Close the recovery chunk manager and kafka consumer.
-      kafkaConsumer.close();
-      chunkManager.stopAsync();
-      chunkManager.awaitTerminated(DEFAULT_START_STOP_DURATION);
-      LOG.info("Finished handling the recovery task: {}", recoveryTaskMetadata);
-      return success;
-    } catch (Exception ex) {
-      LOG.error("Exception in recovery task [{}]: {}", recoveryTaskMetadata, ex);
-      return false;
+
+    var validatedRecoveryTask = validateKafkaOffsets(recoveryTaskMetadata);
```

Comment:
I would avoid using `var` across this PR and use the actual types. We don't use `var` anywhere else in the project today, and it makes it a little tougher to read the code.